include/seqan3/range/view/detail.hpp

```diff
@@ -13,301 +13,296 @@
+ * \tparam right_adaptor_t Type of the second stored adaptor.
+ * \ingroup view
+ *
+ * \details
+ *
+ * If invoked with a range, this adaptor resolves to piping the range into left adaptor and the resulting
+ * range into the right adaptor.
+ */
+template <typename left_adaptor_t, typename right_adaptor_t>
+class combined_adaptor :
+    public adaptor_base<combined_adaptor<left_adaptor_t, right_adaptor_t>,
+                        left_adaptor_t,
+                        right_adaptor_t>
+{
 private:
-    /*!\name Constructors, destructor and assignment
-     * \brief All are declared private to prevent direct use of the CRTP base.
-     * \sa https://isocpp.org/blog/2017/04/quick-q-prevent-user-from-derive-from-incorrect-crtp-base
-     * \{
-     */
-    pipable_adaptor_base() = default;
-    constexpr pipable_adaptor_base(pipable_adaptor_base const &) = default;
-    constexpr pipable_adaptor_base(pipable_adaptor_base &&) = default;
-    constexpr pipable_adaptor_base & operator =(pipable_adaptor_base const &) = default;
-    constexpr pipable_adaptor_base & operator =(pipable_adaptor_base &&) = default;
-    ~pipable_adaptor_base() = default;
-    //!\brief befriend the derived type so that it can instantiate
-    friend derived_type;
-    //!\}
+    //!\brief Type of the CRTP-base.
+    using base_type = adaptor_base<combined_adaptor<left_adaptor_t, right_adaptor_t>,
+                                   left_adaptor_t,
+                                   right_adaptor_t>;
+
+    //!\brief Befriend the base class so it can call impl().
+    friend base_type;
+
+    //!\brief Combine all arguments via `operator|`.
+    template <std::ranges::InputRange urng_t,
+              typename left_adaptor_t_,
+              typename right_adaptor_t_>
+    static auto impl(urng_t && urange, left_adaptor_t_ && left_adaptor, right_adaptor_t_ && right_adaptor)
+    {
+        return std::forward<urng_t>(urange)
+             | std::forward<left_adaptor_t_>(left_adaptor)
+             | std::forward<right_adaptor_t_>(right_adaptor);
+    }
+
+public:
+    //!\brief Inherit the base type's constructors.
+    using base_type::base_type;
+
+    //!\brief Store both arguments in the adaptor.
+    combined_adaptor(left_adaptor_t l, right_adaptor_t r) :
+        base_type{std::forward<left_adaptor_t>(l), std::forward<right_adaptor_t>(r)}
+    {}
 };
 
 // ============================================================================
-//  generic_pipable_view_adaptor
+//  adaptor_for_view_without_args
 // ============================================================================
 
-/*!\brief   A class template template that makes a view pipeable.
+/*!\brief Template for range adaptor closure objects that store no arguments and always delegate to the
+ *        view constructor.
+ * \tparam view_type The view type template.
  * \ingroup view
- * \tparam  view_type The view template that your wish to define the adaptor for.
  *
  * \details
  *
```

Comment:
Can you shortly state why and when you should prefer to inherit from this class instead of `adaptor_base`?